src/functions.cpp

```diff
@@ -2364,7 +2364,58 @@ RCP<const Basic> acoth(const RCP<const Basic> &arg)
 
 RCP<const Basic> ACoth::diff(const RCP<const Symbol> &x) const
 {
-    return mul(div(one, sub(one, pow(x, i2))), get_arg()->diff(x));
+    return mul(div(one, sub(one, pow(get_arg(), i2))), get_arg()->diff(x));
```

Comment:
@sushant-hiray 
Yes, sure. 
I would open a new PR for that after this gets Merged.

@isuruf 
Yes, the test is [here](https://github.com/sympy/csympy/blob/master/src/tests/basic/test_functions.cpp#L1529).